addons/website/controllers/form.py

```diff
@@ -279,6 +279,24 @@ def insert_record(self, request, model, values, custom, meta=None):
                     message_type='comment',
                 )
 
+        if hasattr(record, '_message_log'):
+            authenticate_message = False
+            if request.session.uid:
+                model_record = request.env[model_name]
+                user_email = request.env.user.email
+                form_email = values[model_record._mail_get_primary_email_field()]
+                if user_email != form_email:
+                    authenticate_message = _("This %(model_name)s was submitted by %(user_name)s (%(user_email)s) on behalf of %(form_email)s",
+                        model_name=model.name, user_name=request.env.user.name, user_email=user_email, form_email=form_email)
+            else:
+                authenticate_message = _("/!\\ EXTERNAL SUBMISSION - Customer not verified")
+
+            if authenticate_message:
+                record._message_log(
+                    body=Markup('<div class="alert alert-info" role="alert">{message}</div>').format(message=authenticate_message),
+                    author_id=request.env['ir.model.data']._xmlid_to_res_id('base.partner_root'),
+                )
```

Comment:
Before pinging security team, I just notice, `/!\\` could lead to some errors during the translation, the translator could think the extra `\` is a typo and so could remove it in their translations.

To avoid that eventual issues, we could move that in `Markup` content instead. 🙂 

```suggestion
        if hasattr(record, '_message_log'):
            warning_icon = ""
            authenticate_message = False
            if request.session.uid:
                model_record = request.env[model_name]
                user_email = request.env.user.email
                form_email = values[model_record._mail_get_primary_email_field()]
                if user_email != form_email:
                    authenticate_message = _("This %(model_name)s was submitted by %(user_name)s (%(user_email)s) on behalf of %(form_email)s",
                        model_name=model.name, user_name=request.env.user.name, user_email=user_email, form_email=form_email)
            else:
                warning_icon = "/!\\ "
                authenticate_message = _("EXTERNAL SUBMISSION - Customer not verified")

            if authenticate_message:
                record._message_log(
                    body=Markup('<div class="alert alert-info" role="alert">{warning_icon}{message}</div>').format(warning_icon=warning_icon, message=authenticate_message),
                    author_id=request.env['ir.model.data']._xmlid_to_res_id('base.partner_root'),
                )
```